verification/curator-service/api/src/controllers/auth.ts

```diff
@@ -158,59 +207,119 @@ export class AuthController {
             },
         );
 
+        /**
+         * Generate reset password link
+         */
         this.router.post(
-            '/signup',
-            async (req: Request, res: Response): Promise<void> => {
-                let user = await User.findOne({
-                    email: req.body.email,
-                }).exec();
+            '/request-password-reset',
+            async (req: Request, res: Response): Promise<Response<any>> => {
+                const email = req.body.email as string;
 
-                // Update newsletter preferences
-                if (
-                    user &&
-                    user.newsletterAccepted === false &&
-                    req.body.newsletter
-                ) {
-                    user = await User.findOneAndUpdate(
-                        { email: req.body.email },
-                        { newsletterAccepted: req.body.newsletter },
+                try {
+                    // Check if user with this email address exists
+                    const user = await User.findOne({ email });
+                    if (!user) {
+                        return res.sendStatus(200);
+                    }
+
+                    // Check if there is a token in DB already for this user
+                    const token = await Token.findOne({ userId: user._id });
+                    if (token) {
+                        await token.deleteOne();
+                    }
+
+                    const resetToken = crypto.randomBytes(32).toString('hex');
+                    const hash = await bcrypt.hash(resetToken, 10);
+
+                    // Add new reset token to DB
+                    await new Token({
+                        userId: user._id,
+                        token: hash,
+                        createdAt: Date.now(),
+                    }).save();
+
+                    const resetLink = `http://localhost:3002/reset-password/${resetToken}/${user._id}`;
+
+                    await this.emailClient.send(
+                        [email],
+                        'Password reset link',
+                        `To reset your password click <a href="${resetLink}">here</a>`,
```

Comment:
Won’t resetLink have to be adapted for dev/prod URLs?

Reply:
You're right, I totally forgot to change that, thank you for the catch!